app/models/progression.js

```diff
@@ -20,6 +20,10 @@ class Progression {
   getInitialSubset() {
     return this.model.slice(0, 3).map(item => item.id);
   }
+
+  getDateField(id) {
+    return this.model.find(item => item.id === id).dateField;
+  }
```

Comment:
how are you going to handle `null` scenario? maybe it is worth adding `hasDateField` method?